arch/arm/include/aarch32/cortex_r/stack.h

```diff
@@ -1,47 +1,16 @@
 /*
- * Copyright (c) 2018 Lexmark International, Inc.
+ * Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/**
- * @file
- * @brief Stack helpers for Cortex-R CPUs
- *
- * Stack helper functions.
- */
+#ifndef ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_R_STACK_H_
+#define ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_R_STACK_H_
 
-#ifndef _ARM_CORTEXR_STACK__H_
-#define _ARM_CORTEXR_STACK__H_
-
-#ifdef __cplusplus
-extern "C" {
+#ifndef CONFIG_COMPAT_INCLUDES
+#warning "This header file has moved, include <aarch32/cortex_a_r/stack.h> instead."
```

Comment:
I wonder if this is needed, actually. Do we need this warning, considering the limited group of people currently using cortex-r?

Reply:
> You also need to change the commit title/bodies, to reflect the _a_r_

Fixed

Reply:
Had a similar thought while doing this. The rationale was that the Cortex-R arch support has already been part of three releases and that should warrant a proper deprecation (though, I agree that there would not be that many people affected by this change, especially considering that it has not been all that long since we have changed `arch/arm/cortex_r` to `arch/arm/aarch32/cortex_r`).

If you think this creates more clutter than it helps, I will remove the shim commit.

Reply:
Well, let's leave it, then. When do you think it could be removed? After 2.3 release?

Reply:
Following the API deprecation policy, it should be removed in 2.5; though, this is not really an "API" and two releases sound like quite a long time.

Is there a clearly defined policy for changes like this? or would it be okay to, let's say, just remove it after 2.3?

Reply:
Well I don't think the deprecation policy applies to `arch/`, only to `include/`. The reasoning is that we want to ease the life of application developers, but if you are using arch-specific macros you probably are ready to handle a rename.